utils/s2n_array.c

```diff
@@ -19,6 +19,12 @@
 
 #define S2N_INITIAL_ARRAY_SIZE 16
 
+extern bool s2n_array_is_valid(const struct s2n_array *array)
+{
+    return S2N_OBJECT_PTR_IS_READABLE(array) &&
```

Comment:
surely the array len and element size play into this?

Reply:
They do. We need to check whether the multiplication `len * element_size` doesn't overflow and `mem.size == len * element_size`; however, this blows up the SAT formula, as we assume these values are all nondet. I created the issue https://github.com/awslabs/s2n/issues/2243 to track this.

Reply:
I improved the validity function